java/java/JavaLexer.g4

```diff
@@ -211,8 +211,13 @@ fragment ExponentPart
     : [eE] [+-]? Digits
     ;
 
+fragment EscapeChar
+    : '\\'
+    | '\\u005c'
+    ;
+
 fragment EscapeSequence
-    : '\\' [btnfr"'\\]
+    : EscapeChar [btnfr"'\\]
     | '\\' ([0-3]? [0-7])? [0-7]
     | '\\' 'u'+ HexDigit HexDigit HexDigit HexDigit
```

Comment:
What's the meaning of introducing the new fragment `EscapeChar`? I see only one use-site. Could it be simplified in the following way?

```suggestion
    : '\\' ( 'u005c'? [btnfr"'\\]
           |  ([0-3]? [0-7])? [0-7]
           |  'u'+ HexDigit HexDigit HexDigit HexDigit
           )
```